components/provisioner/internal/installation/components_status.go

```diff
@@ -0,0 +1,83 @@
+package installation
+
+import (
+	"fmt"
+
+	"github.com/kyma-project/control-plane/components/provisioner/internal/model"
+
+	"github.com/kyma-incubator/hydroform/parallel-install/pkg/components"
+	"github.com/kyma-incubator/hydroform/parallel-install/pkg/deployment"
+)
+
+type ComponentsStatus struct {
+	components         map[string]bool
+	componentFailed    error
+	installationFailed error
+}
+
+func NewComponentsStatus(components []model.KymaComponentConfig) *ComponentsStatus {
+	cmpList := make(map[string]bool)
+	for _, component := range components {
+		cmpKey := componentKey(string(component.Component), component.Namespace)
+		cmpList[cmpKey] = false
+	}
+
+	return &ComponentsStatus{components: cmpList}
+}
+
+func (cs *ComponentsStatus) ConsumeEvent(event deployment.ProcessUpdate) {
```

Comment:
During testing when timeout occured for application-connector/xip-patch components the invocation returned nil error with main truncated trace of:

components_status.go:33 
parallel_installation.go:83
parallel_installation.go:99

Reply:
It looks similar to [this issue](https://github.com/kyma-project/control-plane/pull/678#discussion_r654344751). Installer returns event without error inside, e.g.:
```
Installation failed on component bad-component, status: ProcessExecutionFailure, error: <nil>
```
will be reported to the parallel installer team